ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -91,34 +91,49 @@ type Clusters struct {
 	ApiVIP                 string            `yaml:"apiVIP"`
 	IngressVIP             string            `yaml:"ingressVIP"`
 	ClusterName            string            `yaml:"clusterName"`
-	ClusterType            string            `yaml:"clusterType"`
 	AdditionalNTPSources   []string          `yaml:"additionalNTPSources"`
 	Nodes                  []Nodes           `yaml:"nodes"`
 	MachineNetwork         []MachineNetwork  `yaml:"machineNetwork"`
 	ServiceNetwork         []string          `yaml:"serviceNetwork"`
-	NumMasters             uint8             `yaml:"numMasters"`
-	NumWorkers             uint8             `yaml:"numWorkers"`
 	ClusterLabels          map[string]string `yaml:"clusterLabels"`
 	NetworkType            string            `yaml:"networkType"`
 	ClusterNetwork         []ClusterNetwork  `yaml:"clusterNetwork"`
 	IgnitionConfigOverride string            `yaml:"ignitionConfigOverride"`
 	DiskEncryption         DiskEncryption    `yaml:"diskEncryption"`
 	ProxySettings          ProxySettings     `yaml:"proxy,omitempty"`
 	ExtraManifestPath      string            `yaml:"extraManifestPath"`
+
+	NumMasters  uint8
+	NumWorkers  uint8
+	ClusterType string
 }
 
 // Provide custom YAML unmarshal for Clusters which provides default values
 func (rv *Clusters) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	type ClusterDefaulted Clusters
 	var defaults = ClusterDefaulted{
-		ClusterType: SNO,
 		NetworkType: "OVNKubernetes",
-		NumMasters:  1,
 	}
 
 	out := defaults
 	err := unmarshal(&out)
 	*rv = Clusters(out)
+	// Tally master and worker counts based on node roles
+	rv.NumMasters = 0
+	rv.NumWorkers = 0
+	for _, node := range rv.Nodes {
+		if node.Role == "master" {
+			rv.NumMasters += 1
+		} else {
+			rv.NumWorkers += 1
+		}
+	}
+	// Autodetect ClusterType based on the node counts
+	if rv.NumMasters == 1 && rv.NumWorkers == 0 {
```

Comment:
I believe we should raise error if the num of nodes not match requirements similar to AI. 
- standard cluster num of masters must be 3 
- SNO cluster num of master must be 1